pydbtools/utils.py

```diff
@@ -157,7 +156,22 @@ def get_user_id_and_table_dir(
 
 
 def get_database_name_from_userid(user_id: str) -> str:
-    unique_db_name = user_id.split(":")[-1].split("-", 1)[-1].replace("-", "_")
+    """
+    Obtain unique database name for temporary database
+    from various forms of user id
+    """
+    unique_db_name = (
+        # Remove chunk before last ":"
+        user_id.split(":")[-1]
+        # Remove chunk after first "@"
+        .split("@")[0]
+        # Replace remaining dashes with underscores
+        .replace("-", "_")
+    )
+    # Only use permitted characters for AWS Athena databases
+    unique_db_name = "".join(
+        c for c in unique_db_name if c.isalpha() or c.isdigit() or c == "_"
```

Comment:
always appreciate a non-regexy solution for this :) you could condense it as below (though I suppose that isdigit is technically more restrictive than isnumeric; I suspect the chance is pretty low that it'll actually make a difference but I've been around enough to know when not to tempt fate with mad edge cases)
```suggestion
        c for c in unique_db_name if c.isalnum() or c == "_"
```